Traceback (most recent call last):
  File "c:\users\hehez\onedrive\github\pyppg-handbook\venv\lib\site-packages\jupyter_cache\executors\utils.py", line 51, in single_nb_execution
    executenb(
  File "c:\users\hehez\onedrive\github\pyppg-handbook\venv\lib\site-packages\nbclient\client.py", line 1204, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "c:\users\hehez\onedrive\github\pyppg-handbook\venv\lib\site-packages\nbclient\util.py", line 84, in wrapped
    return just_run(coro(*args, **kwargs))
  File "c:\users\hehez\onedrive\github\pyppg-handbook\venv\lib\site-packages\nbclient\util.py", line 62, in just_run
    return loop.run_until_complete(coro)
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.8_3.8.2800.0_x64__qbz5n2kfra8p0\lib\asyncio\base_events.py", line 616, in run_until_complete
    return future.result()
  File "c:\users\hehez\onedrive\github\pyppg-handbook\venv\lib\site-packages\nbclient\client.py", line 663, in async_execute
    await self.async_execute_cell(
  File "c:\users\hehez\onedrive\github\pyppg-handbook\venv\lib\site-packages\nbclient\client.py", line 965, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "c:\users\hehez\onedrive\github\pyppg-handbook\venv\lib\site-packages\nbclient\client.py", line 862, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
# Read ground information
with open("ground_info.json", "r") as read_file:
    ground_data = json.load(read_file)
    
# calculating ground stress and depth of each layer. Returns 'ground_table'
from hapy.shallow_foundation import shallow_footing as sf

(ground_table, u_mid, depth_top,depth_bottom, sig_vo_ef, sig_vo_mid, 
 layer_thickness, midpoint_depth_below_footing, AASHTO_C) = sf.groundinfo(ground_data)

ground_table
------------------

[1;31m---------------------------------------------------------------------------[0m
[1;31mModuleNotFoundError[0m                       Traceback (most recent call last)
Cell [1;32mIn[2], line 6[0m
[0;32m      3[0m     ground_data [38;5;241m=[39m json[38;5;241m.[39mload(read_file)
[0;32m      5[0m [38;5;66;03m# calculating ground stress and depth of each layer. Returns 'ground_table'[39;00m
[1;32m----> 6[0m [38;5;28;01mfrom[39;00m [38;5;21;01mhapy[39;00m[38;5;21;01m.[39;00m[38;5;21;01mshallow_foundation[39;00m [38;5;28;01mimport[39;00m shallow_footing [38;5;28;01mas[39;00m sf
[0;32m      8[0m (ground_table, u_mid, depth_top,depth_bottom, sig_vo_ef, sig_vo_mid, 
[0;32m      9[0m  layer_thickness, midpoint_depth_below_footing, AASHTO_C) [38;5;241m=[39m sf[38;5;241m.[39mgroundinfo(ground_data)
[0;32m     11[0m ground_table

[1;31mModuleNotFoundError[0m: No module named 'hapy'
ModuleNotFoundError: No module named 'hapy'

